### ISSUE TITLE
[Feature Request] Save and import export battery history and show other information such as health

### ISSUE THREAD

**Issue by andrewcrook** [davidwernhart/AlDente#636]
I think it would be great to be able to store and view battery history  e.g 

**So the main data would be ..** 

- cycles
- macOS battery health status 

**You could also add**

- calibrations
- full charge capacity  (mAh)
- temperature of battery?
- see common ranges/averages of discharge charging from and when charged 
   so you can see when stuck in a habit and when to add some variance.

**viewing:**
Would also be useful to see charts and able to compare different data types

**import and export:**
via CSV file
backup
import common formates from third party eg coconutBattery and BatteryHealth 3 etc and restore back up

**other information to view in UI:**
- battery age (days)
- serial 
- manufacture date

Note I think that historical information would help other features finding optimal schedules for calibrations such as 

- [Automatic calibration](https://github.com/davidwernhart/AlDente/issues/572) #572 
- [Ability to schedule Calibration ](https://github.com/davidwernhart/AlDente-Pro-Bug-Tracking/issues/29)

**Comment by itssiggi** [comment]
Yes automatically saving battery health & cycles daily and viewing/exporting the data later on would be really nice!

**Comment by Egregius** [comment]
Any news on this? Would love to have the data stored in a MySQL database for further analysis etc.

**Comment by MatthiasKerbl** [comment]
> Any news on this? Would love to have the data stored in a MySQL database for further analysis etc.

Hi @Egregius,

It's on our internal roadmap but not a priority at the moment.

Best,

Matthias

**Comment by andrewcrook** [comment]
> data stored in a MySQL database

I would recommend using native SQLite/iCloud support. But there could easily be an automated way to export to third-party databases via scripting. The things is, people use many different types of data storage these days.

**Comment by Blackspell01** [comment]
Just curious, if I reset my Mac, everything is gone for now? Or where does coconutBattery save the history?

**Comment by MatthiasKerbl** [comment]
Hi everyone,

The latest version of AlDente (1.30) has a new Dashboard feature that visualizes important battery data.

Best,
Matthias
